datavault/backend.py

```diff
@@ -216,7 +439,379 @@ def hasMore(self, pos):
+    def access(self):
+        self.dataset.attrs['Access Time'] = time.time()
+
+    def getIndependents(self):
+        attrs = self.dataset.attrs
+        rv = []
+        for idx in xrange(sys.maxint):
+            prefix = 'Independent{}.'.format(idx)
+            key = prefix + 'label'
+            if key in attrs:
+                label = attrs[prefix + 'label']
+                shape = attrs[prefix + 'shape']
+                datatype = attrs[prefix + 'datatype']
+                unit = attrs[prefix + 'unit']
+                rv.append(Independent(label, shape, datatype, unit))
+            else:
+                return rv
+
+    def getDependents(self):
+        attrs = self.dataset.attrs
+        rv = []
+        for idx in xrange(sys.maxint):
+            prefix = 'Dependent{}.'.format(idx)
+            key = prefix + 'label'
+            if key in attrs:
+                label = attrs[prefix + 'label']
+                legend = attrs[prefix + 'legend']
+                shape = attrs[prefix + 'shape']
+                datatype = attrs[prefix + 'datatype']
+                unit = attrs[prefix + 'unit']
+                rv.append(Dependent(label, legend, shape, datatype, unit))
+            else:
+                return rv
+
+    def getRowType(self):
+        column_types = []
+        for col in self.getIndependents() + self.getDependents():
+            base_type = col.datatype
+            if base_type in ['v', 'c']:
+                unit_tag = '[{}]'.format(col.unit)
+            else:
+                unit_tag = ''
+            if len(col.shape) > 1:
+                shape_tag = '*{}'.format(len(col.shape))
+                comment = util.braced(','.join(str(s) for s in col.shape))
+            elif col.shape[0] > 1:
+                shape_tag = '*'
+                comment = util.braced(str(col.shape[0]))
+            else:
+                shape_tag = ''
+                comment = ''
+            column_types.append(shape_tag + base_type + unit_tag + comment)
+        type_tag = '*({})'.format(','.join(column_types))
+        return type_tag
+
+    def getTransposeType(self):
+        column_type = []
+        for col in self.getIndependents() + self.getDependents():
+            base_type = col.datatype
+            if base_type in ['v', 'c']:
+                unit_tag = '[{}]'.format(col.unit)
+            else:
+                unit_tag = ''
+            if len(col.shape) > 1:
+                shape_tag = '*{}'.format(len(col.shape) + 1)
+                comment = util.braced('N,' + ','.join(str(s) for s in col.shape))
+            elif col.shape[0] > 1:
+                shape_tag = '*2'
+                comment = util.braced('N,' + str(col.shape[0]))
+            else:
+                shape_tag = '*'
+                comment = ''
+            column_type.append(shape_tag + base_type + unit_tag + comment)
+        type_tag = '({})'.format(','.join(column_type))
+        return type_tag
+
+    def addParam(self, name, data):
```

Comment:
IniData also checks for parameter name collisions and this version doesn't - is that intended?